CONTRIBUTING.md

```diff
@@ -543,10 +533,47 @@ Refer datalad/config.py for information on how to add these environment variable
 - *DATALAD_ALLOW_FAIL*:
   Instructs `@never_fail` decorator to allow to fail, e.g. to ease debugging.
 
-# Changelog section
+# Release(s) workflow
+
+## Branches
+
+- `master`: changes toward the next `MAJOR.MINOR.0` release (e.g., `0.13` as of time of writing).
+  It is also used for release candidates (with `rcX` suffix)
+- `maint`: bug fixes and minor enhancements for the latest released `MAJOR.MINOR.PATCH`
+ (e.g., `0.12.7` as of time of writing)
+- `MAJOR.MINOR`: generally not used, unless some bugfix release with a crticial bugfix is needed.
```

Comment:
If we were to need to do this, I'd vote for a "maint-" in front.

Reply:
On one hand - sounds good. On the other 
- we already have such branches for past releases (well - they could be renamed I guess to follow)
- I thought to keep possibility open for the bright future where we might have a more thorough road map, and decide to accept/review prs for future releases, like already happening, without accumulating them unmerged until release. (For that though we would've need to abandon master/maint for clarity). But hey - if such dire need arises, we indeed could rename branches again.

So, I guess I am ok either way. But it reminded me that if we follow current idea - I should add an explicit statement that changes toward release one after the one to come should aim master but will not be merged until release comes, and should be assigned to the corresponding next milestone

Reply:
Oi, fresh related topic: https://www.google.com/amp/s/www.zdnet.com/google-amp/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/

Reply:
> * we already have such branches for past releases (well - they could be renamed I guess to follow)

I think ideally the old, inactive branches we have (0.5.x, 0.9.x, 0.11.x) should just be deleted.  If we decide for some reason that we need to make a release off of them, we can branch off of the tag.

> * I thought to keep possibility open for the bright future where we might have a more thorough road map, and decide to accept/review prs for future releases, like already happening, without accumulating them unmerged until release.

If I understand your suggestion, it's something I'd be opposed to, but that's a separate proposal and discussion.

However, re "like already happening": We (in particular me) botched this rc.  I didn't realize you were opposed to releasing `push` in its current form.  We shouldn't have entered the rc phase when things were not settled internally.  It makes what should be a short period drag on indefinitely.

Reply:
> However, re "like already happening": 

I had in mind PRs like #4597 which are to be  "scheduled" for the next release

<details>
<summary>I think the `push` part is not really related to this PR thus collapsed here</summary>

> We (in particular me) botched this rc. I didn't realize you were opposed to releasing push in its current form.

Hm... you can prove me wrong, but I thought that I never actively "opposed" - that is why the only issue in 0.13.0 milestone ATM is https://github.com/datalad/datalad/pull/4453 but I will remove it from any milestone since I do not think we have reached or will reach any agreement on that any time soon and it is not really a critical issue.  I did express my dislikes though, and stated that we would likely need to break API if we release `push` in current form.  So I think it was just a common sense and delaying the release while we hadn't agreed on the API.

</details>

Reply:
> I had in mind PRs like #4597 which are to be "scheduled" for the next release

Yes.  My point is that it's being held up by an rc that should not have taken as long as it has.  And the reason it is taking so long is we entered the rc period without realizing there disagreements about `push` in the form it was in at rc1.

Reply:
I would still vote for no `maint-` prefix for the reasons stated above. So we need one more vote from other @datalad/developers to break the tie.

Reply:
or -- on a 2nd though -- "whatever!  + we do have `maint` branch used actively , so ok -- let's go with `maint-` prefix". adjusting